openedx/core/djangoapps/django_comment_common/comment_client/models.py

```diff
@@ -208,3 +215,169 @@ def url(cls, action, params=None):
+            "course_key": course_id
+        }
+        request_data = {k: v for k, v in request_data.items() if v is not None}
+        response = forum_api.update_thread(**request_data)
+        return response
+
+    def perform_http_put_request(self, request_params):
+        url = self.url(action="put", params=self.attributes)
+        response = perform_request(
+            "put",
+            url,
+            request_params,
+            metric_tags=self._metric_tags,
+            metric_action="model.update",
+        )
+        return response
+
+    def perform_http_post_request(self):
+        url = self.url(action="post", params=self.attributes)
+        response = perform_request(
+            "post",
+            url,
+            self.initializable_attributes(),
+            metric_tags=self._metric_tags,
+            metric_action="model.insert",
+        )
+        return response
+
+    def handle_create(self, params=None):
+        course_id = self.attributes.get("course_id") or params.get("course_id")
+        course_key = get_course_key(course_id)
+        if is_forum_v2_enabled(course_key):
+            response = None
+            if self.type == "comment":
+                response = self.handle_create_comment(str(course_key))
+            elif self.type == "thread":
+                response = self.handle_create_thread(str(course_key))
+            if response is None:
+                raise CommentClientRequestError("Forum v2 API call is missing")
+        else:
+            response = self.perform_http_post_request()
+        return response
+
+    def handle_create_comment(self, course_id):
+        request_data = self.initializable_attributes()
+        try:
+            body = request_data["body"]
+            user_id = request_data["user_id"]
+            course_id = course_id or str(request_data["course_id"])
+        except KeyError as e:
+            raise e
+        if parent_id := self.attributes.get("parent_id"):
+            response = forum_api.create_child_comment(
+                parent_id,
+                body,
+                user_id,
+                course_id,
+                request_data.get("anonymous", False),
+                request_data.get("anonymous_to_peers", False),
+            )
+        else:
+            response = forum_api.create_parent_comment(
+                self.attributes["thread_id"],
+                body,
+                user_id,
+                course_id,
+                request_data.get("anonymous", False),
+                request_data.get("anonymous_to_peers", False),
+            )
+        return response
+
+    def handle_create_thread(self, course_id):
+        request_data = self.initializable_attributes()
+        try:
+            title = request_data["title"]
+            body = request_data["body"]
+            user_id = str(request_data["user_id"])
+        except KeyError as e:
+            raise e
```

Comment:
This is raising an issue on my computer:

```
...
lms-1  |   File "/openedx/edx-platform/lms/djangoapps/discussion/rest_api/views.py", line 667, in create
lms-1  |     return Response(create_thread(request, request.data))
lms-1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lms-1  |   File "/openedx/edx-platform/lms/djangoapps/discussion/rest_api/api.py", line 1482, in create_thread
lms-1  |     serializer.save()
lms-1  |   File "/openedx/venv/lib/python3.11/site-packages/rest_framework/serializers.py", line 212, in save
lms-1  |     self.instance = self.create(validated_data)
lms-1  |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
lms-1  |   File "/openedx/edx-platform/lms/djangoapps/discussion/rest_api/serializers.py", line 467, in create
lms-1  |     thread.save()
lms-1  |   File "/openedx/edx-platform/openedx/core/djangoapps/django_comment_common/comment_client/models.py", line 169, in save
lms-1  |     response = self.handle_create(params)
lms-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^
lms-1  |   File "/openedx/edx-platform/openedx/core/djangoapps/django_comment_common/comment_client/models.py", line 325, in handle_create
lms-1  |     response = self.handle_create_thread(str(course_key))
lms-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lms-1  |   File "/openedx/edx-platform/openedx/core/djangoapps/django_comment_common/comment_client/models.py", line 367, in handle_create_thread
lms-1  |     raise e
lms-1  |   File "/openedx/edx-platform/openedx/core/djangoapps/django_comment_common/comment_client/models.py", line 365, in handle_create_thread
lms-1  |     user_id = str(request_data["user_id"])
lms-1  |                   ~~~~~~~~~~~~^^^^^^^^^^^
lms-1  | KeyError: 'user_id'
```

Why are we catching the KeyError and raising it again?

Reply:
yeah point, we can just remove the try catch.
Actually, might be forum master branch is not that stable yet. 
We tested the flow with forum v2(on this branch: feat/add_tests_for_forum_v2_native_python_calls).
But this'll also fail now as we have changed the signatures for native APIs.

Reply:
Also the backend is set to mysql in forum master branch.
Plus let me see this user_id issue, this shouldn't occur as it is a required field.

Reply:
actually i did fix this error [here](https://github.com/edly-io/forum/pull/100/commits/c8af1504e3332665712175b755e09c8667856655). 
made a new PR [here](https://github.com/edly-io/forum/pull/105).
User_id issue will be resolved after merge.